admin/includes/modules/copy_product_confirm.php

```diff
@@ -47,52 +41,33 @@
                     SET products_quantity_order_units = 1
                     WHERE products_id = " . $products_id;
             $results = $db->Execute($sql);
+            $product->fields['products_quantity_order_units'] = 1; 
         }
         // fix Product copy from if Minimum is 0
         if ($product->fields['products_quantity_order_min'] == 0) {
             $sql = "UPDATE " . TABLE_PRODUCTS . "
                     SET products_quantity_order_min = 1
                     WHERE products_id = " . $products_id;
             $results = $db->Execute($sql);
+            $product->fields['products_quantity_order_min'] = 1; 
         }
 
-        $tmp_value = zen_db_input($product->fields['products_quantity']);
-        $products_quantity = (!zen_not_null($tmp_value) || $tmp_value == '' || $tmp_value == 0) ? 0 : $tmp_value;
-        $tmp_value = zen_db_input($product->fields['products_price']);
-        $products_price = (!zen_not_null($tmp_value) || $tmp_value == '' || $tmp_value == 0) ? 0 : $tmp_value;
-        $tmp_value = zen_db_input($product->fields['products_weight']);
-        $products_weight = (!zen_not_null($tmp_value) || $tmp_value == '' || $tmp_value == 0) ? 0 : $tmp_value;
```

Comment:
I'd like to see this change still accommodate specific handling for `float`/`decimal` fields like these (and also for `integer` fields), particularly when the incoming input is left blank; otherwise you'll have SQL Strict errors.

Reply:
Roger that.  Will do shortly.